src/charm.py

```diff
@@ -159,47 +67,26 @@ def _on_publish_relation_changed(self, event):
         self._reload_config()
 
     def _on_config_changed(self, _):
-        for key in self.model.config:
-            if key not in self._stored.config:
-                value = self.model.config[key]
-                self._stored.config[key] = value
-            if self.model.config[key] != self._stored.config[key]:
-                value = self.model.config[key]
-                logger.info("Setting {} to: {}".format(key, value))
-                self._stored.config[key] = value
-
-        # this should go into install hook
-        uid = pwd.getpwnam("root").pw_uid
-        gid = grp.getgrnam("root").gr_gid
-
-        path = "/etc/nginx/ssl"
-        if not os.path.exists(path):
-            os.makedirs(path, 0o755)
-            os.chown(path, uid, gid)
-            os.chmod(path, 0o755)
-
-        config = self._stored.config
-        self._stored.config["ssl_enabled"] = (
-            "ssl_cert" in config.keys()  # noqa: W503
-            and "ssl_key" in config.keys()  # noqa: W503
-            and "ssl_cert" in self._stored.config.keys()  # noqa: W503
-            and "ssl_key" in self._stored.config.keys()  # noqa: W503
-        )
+        config = self.model.config
+        for key in config:
+            self._stored.config[key] = config[key]
 
-        # write ssl certificate if present
-        if "ssl_cert" in config.keys() and self._stored.config["ssl_cert"]:
-            cert_path = os.path.join("/etc/nginx/ssl", "server.crt")
-            cert = b64decode(self._stored.config["ssl_cert"])
-            write_file(cert_path, cert, "root", "root", 0o644)
-
-        if "ssl_key" in config.keys() and self._stored.config["ssl_key"]:
-            key_path = os.path.join("/etc/nginx/ssl", "server.key")
-            key = b64decode(self._stored.config["ssl_key"])
-            write_file(key_path, key, "root", "root", 0o640)
-
-        if "ssl_ca" in config.keys() and self._stored.config["ssl_ca"]:
-            ca_cert = self._stored.config["ssl_ca"]
-            install_ca_cert(b64decode(ca_cert), "nginx-server.crt")
+        try:
+            if config.get("ssl_cert"):
+                cert_path = os.path.join("/etc/nginx/ssl", "server.crt")
+                cert = b64decode(config["ssl_cert"])
+                write_file(cert_path, cert, 0o644)
```

Comment:
```suggestion
                write_file("/etc/nginx/ssl/server.crt", b64decode(config["ssl_cert"]), 0o644)
```

These lines can be simplified.  Same for the next block.